Backtracking/generateParentheses.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Problem Statement: Given a number n pairs of parenthses, try to Generate all combinations of valid parentheses
+ *
+ * link to problem: https://leetcode.com/problems/generate-parentheses/
+ *
+ */
+
+const generateParentheses = (n, chres, openParenthese, closedParenthese, res) => {
+  if (openParenthese === n && closedParenthese === n) {
+    res.push(chres)
+    return
+  }
+
+  if (openParenthese <= n) {
+    generateParentheses(n, chres + '(', openParenthese + 1, closedParenthese, res)
+  }
+
+  if (closedParenthese < openParenthese) {
+    generateParentheses(n, chres + ')', openParenthese, closedParenthese + 1, res)
+  }
+}
+
+const res = []
+generateParentheses(3, [], 0, 0, res)
```

Comment:
Should be in a proper test.

Reply:
can I not test the function, because it is a void function it just fills the res variable with the result ?

Reply:
Done, I added a proper Jtest